### DISCUSSION TITLE
Can I replace store.Store with a custom version with persistent cache?

### DISCUSSION THREAD

**Discussion by Ilyes512** [pocketbase/pocketbase#3633] [Q&A]
First of, this is my first post and I must say PocketBase looks awesome! Nice work :)

So I was looking at the code and wondered if I could create a `pocketbase backup`-subcommand. And in my quick glance I came across the usage of the (cache) store for "blocking" concurrent usage of the backup logic. For instance see: https://github.com/pocketbase/pocketbase/blob/master/core/base_backup.go#L47

So I would like to be able to run the command against a running (`pocketbase serve`) instance. Because the store uses in-memory storage it won't actually prevent you from running the backup command if at that same moment a backup is running using the UI, or the API or even the auto-backup feature.

So thats why I am looking at replacing the store.Store implementation with something like bbolt or maybe even just use the sqlite db.

Using persistent cache does come with some challenges like for example, what will happen if the app crashes and it does not remove the persisted lock. I could use some kind of TTL for the cache value, maybe?

So any advice or ideas?

Of course I could just not enable auto-backups and not use the backup api and take the risk that someone might use the manual backups using the UI at the same time. Actually this would also not prevent write/reads 

(Just for some context... I am just "playing" around with PocketBase and getting to learn the code base and the tool it self for fun at the moment.)

**Comment by ganigeorgiev** [comment]
The persistence of `app.Cache()` (_will be renamed/soft-deprecated in the future in favour of `app.Store()`_) is not customizable at the moment.

I've added it in my todo to consider accommodating backups initialization from different processes, but for now this remains a very low priority.
